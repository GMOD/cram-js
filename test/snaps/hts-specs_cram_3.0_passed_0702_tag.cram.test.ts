import { testFile } from '../testUtil'
testFile('hts-specs/cram/3.0/passed/0702_tag.cram')
