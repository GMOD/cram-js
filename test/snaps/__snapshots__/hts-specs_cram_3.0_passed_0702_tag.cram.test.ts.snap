// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole hts-specs/cram/3.0/passed/0702_tag.cram without error 1`] = `
[
  {
    "fileId": "_.cram",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 3571883735,
      "landmarks": [
        0,
        160,
      ],
      "length": 249,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 3571883735,
        "landmarks": [
          0,
          160,
        ],
        "length": 249,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/CRAM/passed/ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/CRAM/passed/ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              40,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 32,
          },
          "parametersBytes": 1,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {
        "CVI": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  4,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 4413001,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "MeZ": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 5072218,
            "stopByte": 9,
          },
          "parametersBytes": 5,
        },
        "MpZ": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 5075034,
            "stopByte": 9,
          },
          "parametersBytes": 5,
        },
        "PIf": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  4,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 5261670,
              },
              "parametersBytes": 4,
            },
          },
          "parametersBytes": 12,
        },
        "Z0Z": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 5910618,
            "stopByte": 9,
          },
          "parametersBytes": 5,
        },
      },
      "tagIdsDictionary": [
        [
          "CVI",
          "PIf",
        ],
        [
          "Z0Z",
          "MeZ",
        ],
        [
          "Z0Z",
          "MeZ",
          "MpZ",
        ],
      ],
    },
    "containerHeader": {
      "_endPosition": 315,
      "_size": 21,
      "alignmentSpan": 300,
      "crc32": 2528032824,
      "landmarks": [
        231,
      ],
      "length": 973,
      "numBases": 400,
      "numBlocks": 14,
      "numLandmarks": 1,
      "numRecords": 4,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1000,
    },
    "data": [
      {
        "_endPosition": 315,
        "_size": 21,
        "alignmentSpan": 300,
        "crc32": 2528032824,
        "landmarks": [
          231,
        ],
        "length": 973,
        "numBases": 400,
        "numBlocks": 14,
        "numLandmarks": 1,
        "numRecords": 4,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1000,
      },
      {
        "features": [
          {
            "alignmentStart": 1000,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "readName": "r1",
              "sequenceId": 0,
              "uniqueId": 554,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "r1",
            "sequenceId": 0,
            "tags": {
              "CV": 299792458,
              "PI": 3.141590118408203,
            },
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 552,
          },
          {
            "alignmentStart": 1000,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "readName": "r2",
              "sequenceId": 0,
              "uniqueId": 555,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "r2",
            "sequenceId": 0,
            "tags": {
              "CV": 299792458,
              "PI": 3.141590118408203,
            },
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 553,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1000,
              "readName": "r1",
              "sequenceId": 0,
              "uniqueId": 552,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "r1",
            "sequenceId": 0,
            "tags": {
              "Me": "9.1093837015x10^-31 kg",
              "Z0": "299792458 Ohms",
            },
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 554,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1000,
              "readName": "r2",
              "sequenceId": 0,
              "uniqueId": 553,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "r2",
            "sequenceId": 0,
            "tags": {
              "Me": "9.1093837015x10^-31 kg",
              "Mp": "1.67262192370x10^-27 kg",
              "Z0": "299792458 Ohms",
            },
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 555,
          },
        ],
        "header": {
          "_endPosition": 611,
          "_size": 60,
          "compressedSize": 56,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            131,
            232,
            129,
            44,
            4,
            0,
            12,
            11,
            11,
            12,
            15,
            16,
            17,
            32,
            224,
            67,
            86,
            73,
            224,
            90,
            48,
            90,
            224,
            77,
            101,
            90,
            224,
            77,
            112,
            90,
            224,
            80,
            73,
            102,
            255,
            255,
            255,
            255,
            15,
            188,
            14,
            187,
            152,
            12,
            130,
            56,
            146,
            25,
            54,
            223,
            160,
            201,
            234,
            161,
            96,
          ],
          "contentId": 0,
          "contentPosition": 551,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2577998168,
          "parsedContent": {
            "_endPosition": 371,
            "_size": 56,
            "contentIds": [
              11,
              12,
              15,
              16,
              17,
              32,
              4413001,
              5910618,
              5072218,
              5075034,
              5261670,
            ],
            "md5": [
              188,
              14,
              187,
              152,
              12,
              130,
              56,
              146,
              25,
              54,
              223,
              160,
              201,
              234,
              161,
              96,
            ],
            "numBlocks": 12,
            "numContentIds": 11,
            "numRecords": 4,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 300,
            "refSeqStart": 1000,
          },
          "uncompressedSize": 56,
        },
      },
    ],
  },
]
`;
